Frameworks/CoreGraphics/CGColorSpace.mm

```diff
@@ -38,135 +39,148 @@
+    return __CGColorSpace::CreateInstance(kCFAllocatorDefault, model);
 }
 
 /**
  @Status Stub
- @Notes Always returns RGB colorspace
 */
-CGColorSpaceRef CGColorSpaceCreatePattern(CGColorSpaceRef source) {
-    UNIMPLEMENTED();
-    return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelPattern);
+CGColorSpaceRef CGColorSpaceCreateCalibratedRGB(const CGFloat whitePoint[3],
+                                                const CGFloat blackPoint[3],
+                                                const CGFloat gamma[3],
+                                                const CGFloat matrix[9]) {
+    UNIMPLEMENTED_WITH_MSG("CGColorSpaceCreateCalibratedRGB not supported");
+    return __CGColorSpace::CreateInstance(kCFAllocatorDefault, kCGColorSpaceModelRGB);
 }
 
 /**
  @Status Interoperable
 */
-CGColorSpaceRef CGColorSpaceCreateDeviceGray() {
-    return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelMonochrome);
-}
-
-CGColorSpaceRef _CGColorSpaceCreate(CGColorSpaceModel model) {
-    return static_cast<CGColorSpaceRef>(new __CGColorSpace(model));
+CGColorSpaceRef CGColorSpaceCreateIndexed(CGColorSpaceRef baseSpace, size_t lastIndex, const unsigned char* colorTable) {
+    RETURN_NULL_IF(!baseSpace);
+    RETURN_NULL_IF(lastIndex > 255);
+    RETURN_NULL_IF(!colorTable);
+    return __CGColorSpace::CreateInstance(kCFAllocatorDefault, kCGColorSpaceModelIndexed, baseSpace, lastIndex, colorTable);
 }
 
 /**
  @Status Caveat
  @Notes Only GenericRGB, GenericRGBLinear and GenericGray supported
 */
 CGColorSpaceRef CGColorSpaceCreateWithName(CFStringRef name) {
-    const char* strName = static_cast<const char*>([(__bridge NSString*)name UTF8String]);
-
-    if (strcmp(strName, "kCGColorSpaceGenericRGB") == 0) {
-        return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelRGB);
-    } else if (strcmp(strName, "kCGColorSpaceGenericRGBLinear") == 0) {
-        return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelRGB);
-    } else if (strcmp(strName, "kCGColorSpaceGenericGray") == 0) {
-        return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelMonochrome);
+    RETURN_NULL_IF(!name);
+    if (CFEqual(kCGColorSpaceGenericRGB, name)) {
+        return __CGColorSpace::CreateInstance(kCFAllocatorDefault, kCGColorSpaceModelRGB);
+    } else if (CFEqual(kCGColorSpaceGenericRGBLinear, name)) {
+        return __CGColorSpace::CreateInstance(kCFAllocatorDefault, kCGColorSpaceModelRGB);
+    } else if (CFEqual(kCGColorSpaceGenericGray, name)) {
+        return __CGColorSpace::CreateInstance(kCFAllocatorDefault, kCGColorSpaceModelMonochrome);
     } else {
-        UNIMPLEMENTED_WITH_MSG("Colorspace Unsupported: %s", strName);
-        return (CGColorSpaceRef) new __CGColorSpace(kCGColorSpaceModelRGB);
+        UNIMPLEMENTED_WITH_MSG("Colorspace Unsupported: %s", CFStringGetCStringPtr(name, kCFStringEncodingUTF8));
+        return __CGColorSpace::CreateInstance(kCFAllocatorDefault, kCGColorSpaceModelRGB);
```

Comment:
return CGColorSpaceCreateDeviceRGB() here so we get shared const value rather than creating new